Bottom: 2aad87442f67375a1d96704ceb15db2efd6f265d
Top:    32ed1618c1127e08240be10deb4d0d448bd5cefc
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-07-16 15:13:50 -0500

Create new role buildbot-worker-assets

Starts a web server for distributing assets to buildbot workers.


---
diff --git a/ansible-playbook/buildbot.yml b/ansible-playbook/buildbot.yml
index f5bf035e..7e8eccf0 100644
--- a/ansible-playbook/buildbot.yml
+++ b/ansible-playbook/buildbot.yml
@@ -26,6 +26,20 @@
       new_user_email: buildbot@novatech-llc.com
       new_user_pass: "{{ lookup('password', '/dev/null') }}"
 
+- name: Setup Buildbot worker system
+  hosts: builder
+  become: true
+  become_user: root
+  become_method: sudo
+  tags:
+  - buildbot
+  - buildbot-worker
+  vars:
+    asset_host: "{{ hostvars['buildbot-assets'].uri }}"
+    buildbot_master: "{{ hostvars.buildbot_master.ip_addr }}"
+  roles:
+  - role: buildbot-worker-assets
+
 - name: Gitlab public keys for buildbot containers
   hosts: daedalus
   become: true
diff --git a/roles/buildbot-worker-assets/tasks/main.yml b/roles/buildbot-worker-assets/tasks/main.yml
new file mode 100644
index 00000000..8f602634
--- /dev/null
+++ b/roles/buildbot-worker-assets/tasks/main.yml
@@ -0,0 +1,17 @@
+---
+- name: Start asset server
+  docker_container:
+    name: buildbot-assets
+    image: "httpd:alpine"
+    volumes:
+    - "/mnt/assets:/usr/local/apache2/htdocs/:ro"
+    ports:
+    - 80
+  register: asset_server
+
+- add_host:
+    name: buildbot-assets # required. The hostname/ip of the host to add to the inventory, can include a colon and a port number.
+    uri: "http://\
+      {{asset_server.ansible_facts.docker_container.NetworkSettings.Gateway}}\
+      :{{asset_server.ansible_facts.docker_container.NetworkSettings.Ports['80/tcp'][0].HostPort}}\
+      "
